minimint/src/lib.rs

```diff
@@ -155,6 +156,21 @@ impl MinimintServer {
         proposal: impl Future<Output = ConsensusProposal>,
         rng: &mut (impl RngCore + CryptoRng + Clone + 'static),
     ) -> Vec<ConsensusOutcome> {
+        // for testing federations with one peer
+        if self.cfg.peers.len() == 1 {
+            tokio::select! {
+              () = self.consensus.transaction_notify.notified() => (),
+              () = self.consensus.await_consensus_proposal() => (),
+            }
+            let proposal = proposal.await;
+            let epoch = self.hbbft.next_epoch() + 1;
+            self.hbbft.skip_to_epoch(epoch);
```

Comment:
Did you check if HBBFT might already support fake 1-member federations?

Reply:
Yeah, it never returns an outcome.